unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/ChatWidgetHUD/ChatHUDView.cs

```diff
@@ -452,13 +454,23 @@ private void SortEntriesImmediate()
         {
             if (this.entries.Count <= 0) return;
 
-            var entries = this.entries.Values.OrderBy(x => x.Model.timestamp).ToList();
-
-            for (var i = 0; i < entries.Count; i++)
+            if (UseLegacySorting)
             {
-                if (entries[i].transform.GetSiblingIndex() != i)
-                    entries[i].transform.SetSiblingIndex(i);
+                var entries = this.entries.Values.OrderBy(x => x.Model.timestamp).ToList();
+
+                for (var i = 0; i < entries.Count; i++)
+                {
+                    if (entries[i].transform.GetSiblingIndex() != i)
+                        entries[i].transform.SetSiblingIndex(i);
+                }
+
+                return;
             }
```

Comment:
What's the point of the sorting strategy if we are gonna have this hardcoded? 

Instead of a IComparer, your can solve it converting SortingStrategy into an `IEnumerator<>` that returns you the entries in whichever order. And inside you can do anything you want.

Reply:
Sorry I meant an `IEnumerable<>`. something like: `Func<IEnumerable, any params you need to sort>` would work. Anyway, approving to avoid blocking